src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1550,10 +1550,6 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                     scriptConfig.Functions.Add((string)function);
                 }
             }
-            else
-            {
-                scriptConfig.Functions = null;
```

Comment:
The test was flaking out because while it was specifying a functions whitelist, the whitelist wasn't actually being applied. There were then 4 functions running, and from time to time the queue function might get a message and timeout as well, leading to the flakiness. The test was verifying only a **single** timeout exception and there were multiple occurring.

Reply:
ApplyConfiguration_ClearsFunctionsFilter test failed in the PR build. Please take a look.